js7/js_hw7_t2.js

```diff
@@ -0,0 +1,52 @@
+
+const characters = [
+    { 'name': 'Barney', 'age': 36 },
+    { 'name': 'Fred', 'age': 40 },
+    { 'name': 'Jack', 'age': 50 }
+  ];
+
+
+// 1. Напишите функцию addCharacter(character) позволяющую добавить новый объект в массив characters. 
+//     Объект должен иметь поля name (string) и age (number)
+
+function addCharacter(character = {}) {
+    if ('name' in character && 'age' in character && typeof character['name'] === 'string' && typeof character['age'] === 'number') {
+     return characters.push(character);
+       } 
+}
+addCharacter({'name': 'Angel', 'age': 27});
+console.log(characters);
+
+// 2. Напишите функцию getCharacter(name), позволяющую получить объект персонажа по его имени// getCharacter('Fred') => { 'name': 'Fred', 'age': 40 }
+function getCharacter(nameChar = ''){
+    if (characters.map(element => Object.values(element)).flat().includes(nameChar)) {
+    return characters.find(character => character.name === nameChar);
+    } 
```

Comment:
Не совсем понятно для чего нам в условии нужем массив, ведь `map` возвращает именно его и даже если он будет пуст, то в условии будет `true`
все же проще сразу искать и если не нашли, соответствующую ошибку

```suggestion
    const character = characters.find(character => character.name === nameChar);
    if (!character) {
       console.log('Not found');
     } else {
         return character;
      }
```